lib/instrumentation/langchain/common.js

```diff
@@ -43,11 +43,17 @@ common.mergeMetadata = function mergeMetadata(localMeta = {}, paramsMeta = {}) {
  *
  * @param {object} params function params
  * @param {Agent} params.agent NR agent
+ * @param {Shim} params.shim current shim instance
  * @param {string} params.type type of llm event(i.e.- LlmChatCompletionMessage, LlmTool, etc)
  * @param {object} params.msg the llm event getting enqueued
  * @param {string} params.pkgVersion version of langchain library instrumented
  */
-common.recordEvent = function recordEvent({ agent, type, msg, pkgVersion }) {
+common.recordEvent = function recordEvent({ agent, shim, type, msg, pkgVersion }) {
+  if (common.shouldSkipInstrumentation(agent.config) === true) {
```

Comment:
this behavior is different than the other libraries so when ai_monitoring.enabled is false it still creates the events in memory but doesn't do anything with them.

Reply:
there are cases where it is creating LLM events like `LlmErrorMessage`. It's also doing a lot of parsing. I would change langchain instrumentation and check for this flag before it attempts to event create any llm events